backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java

```diff
@@ -66,13 +66,13 @@ public void clearAllTables() {
                 .block();
     }
 
-    private Mono<?> truncate(CassandraAsyncExecutor executor, String name) {
+    private Mono<Void> truncate(CassandraAsyncExecutor executor, String name) {
         return executor.executeReactor(
                 QueryBuilder.select()
                         .from(name)
                         .limit(1)
                         .setFetchSize(1))
                 .filter(resultSet -> !resultSet.isExhausted())
-                .thenEmpty(executor.executeVoidReactor(QueryBuilder.truncate(name)));
```

Comment:
could you explain why this construct is better ?

Reply:
It's not better, it's the right way to do it, `then*` waits for the completion of the previous `Publisher` and evaluate whether is was empty or not, while `*map` evaluate it only if the previous one have a value.